mypy/fastparse.py

```diff
@@ -213,10 +214,56 @@ def parse(
         assert options.python_version[0] >= 3
         feature_version = options.python_version[1]
     try:
+        ignore_comment_errors = options and options.ignore_comment_errors
+
+        def try_parse_ast(source: str) -> tuple[AST | None, str]:
+            try:
+                ast = ast3_parse(source, fnam, "exec", feature_version=feature_version)
+            except SyntaxError as e:
+
+                def strip_comment(line: str) -> str:
+                    return re.sub(r"([^#]*).*(\r\n|\r|\n)?", r"\1\2", line)
+
+                def strip_comment_from_line(text: str, target_lineno: int) -> str:
+                    lines = text.splitlines(keepends=True)
+                    return "".join(
+                        line if lineno != target_lineno else strip_comment(line)
+                        for lineno, line in enumerate(lines, start=1)
+                    )
+
+                if (
+                    ignore_comment_errors
+                    and e.text
+                    and e.lineno
+                    and TYPE_COMMENT_PATTERN.match(e.text)
+                ):
+                    stripped_source = strip_comment_from_line(source, e.lineno)
+                    errors.report(
+                        e.lineno or -1,
+                        e.offset or -1,
+                        # TODO: It would be nicer if the message included the line text via
+                        # `e.text.strip()`. However, I can't figure out how to make the unit
+                        # test pass, because the # N: <expected_result> in the unit test must
+                        # contain itself. If there was some partial matching mechanism for the
+                        # unit test comment, then we could make this work.
```

Comment:
In this case, an `[out]` section containing the exact output can be easier to write than inline assertions.

Reply:
Thanks, I'll look into that. I'm assuming that `[out]` sections are described somewhere in the testing README.

Reply:
Actually it turns out that - yes :) it's [documented](https://github.com/python/mypy/blob/master/test-data/unit/README.md), though I mostly learned it by example.

> - optional `[out]` is an alternative to the `# E: ` notation: it indicates that
any text after it contains the expected type checking error messages.
Usually, `# E: ` is preferred because it makes it easier to associate the
errors with the code generating them at a glance, and to change the code of
the test without having to change line numbers in `[out]`